src/plugins/intel_cpu/src/nodes/executors/fullyconnected_implementations.cpp

```diff
@@ -0,0 +1,301 @@
+                    dnnl::impl::memory_desc_wrapper wrapped(dnnlDesc->getDnnlDesc().get());
+                    return wrapped.offset0();
+                };
+
+                VERIFY(dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core), UNSUPPORTED_ISA);
+                VERIFY(srcType(config) == ov::element::f32, UNSUPPORTED_SRC_PRECISIONS);
+                // disable rank=4:
+                // if layout is nhwc:
+                //   A matrix: N * IC * H * W --> N * (IC*H*W), the M, N', K of matrix multiply will be:
+                //   M = 1, K = (IC*H*W), when M = 1 it should not be efficient since acts as a vector multiply
+                // if layout is nchw/nChw16c: brg1x1 not support. Although jit supports, it should have similar
+                //   problems with the above.
+                VERIFY(one_of(srcRank(config), 2u, 3u), UNSUPPORTED_SRC_RANK);
+                VERIFY(weiRank(config) == 2, UNSUPPORTED_WEI_RANK);
+                // brg convolution does not support stride
+                VERIFY(getOffset0(config.descs.at(ARG_DST)) == 0, UNSUPPORTED_DST_STRIDES);
+                return true;
+            },
+            // requiresFallback
+            [](const FCConfig& config) -> ov::optional<executor::Config<FCAttrs>> {
+                // @todo use dnnlConvolutionLayoutConfig after one is implemented
+                return requiresFallbackCommon(config, dnnlConvolutionTypeMapping, dnnlFCLayoutConfig, dnnlFCMappingNotation);
+            },
+            // acceptsShape
+            [](const MemoryArgs& memory) -> bool {
+                const auto inRank = memory.at(ARG_SRC)->getShape().getRank();
+                const auto& inDims = memory.at(ARG_SRC)->getShape().getDims();
+                const auto& weightDims = memory.at(ARG_WEI)->getShape().getDims();
+                // for original inner product semantics:
+                //  when input is 2D tensor -> M in oneDNN will map to widthInConv
+                //  when input is 3D tensor -> M in oneDNN will map to widthInConv*minibatch
+                // currently nwc mapping in brg::
+                //  when input is 2D tensor -> widthInConv will map to 'w', 'n' will be 1
+                //  when input is 3D tensor -> widthInConv will map to 'w', 'n' will be minibatch
+                Dim widthInConv = inDims[inRank - 2];
+                Dim K = inDims[inRank - 1];
+                Dim N = weightDims[0];
+
+                const auto& weightsSize = memory.at(ARG_WEI)->getDesc().getCurrentMemSize();
+                // Disable Conv1x1 when weight size >= 16M to avoid different weight layout when having different input
+                // activation shapes. As a consuquence, peak memory consumption in LLM can be decreased.
+                VERIFY(weightsSize < (16 * 1 << 20), " weights size is to big");
+                VERIFY(widthInConv >= 2 && widthInConv <= 3136 && K >= 96 && K <= 4096 && N >= 96 && N <= K * 4,
```

Comment:
As I can see `acceptsShape` is called inside `preconfigure` on createPrimitive stage. Given that shapes still might be dynamic on this level. Am I missing smt?

Reply:
You are right.
The problem here is we do want to select an executor in scope of preconfigure call, as an assumption about the executor which will be selected during the inference.
But at the preconfigure stage the shape is not known, so we have to deal with the limited amount information (shapes might be dynamic as you mentioned)

Reply:
Added a leftover